dask/array/routines.py

```diff
@@ -410,11 +410,26 @@ def round(a, decimals=0):
     return a.map_blocks(np.round, decimals=decimals, dtype=a.dtype)
 
 
+def _unique_return_index(parts, chunks):
+    arr = np.concatenate([p[0] for p in parts])
+    offset = np.cumsum((0, ) + chunks)[:-1]
+    idx = np.concatenate([p[1] + o for (p, o) in zip(parts, offset)])
+
+    u, i = np.unique(arr, return_index=True)
+    return u, idx[i]
```

Comment:
Given that we only use this function once it might make sense to just include it in the function directly.

Reply:
Any thoughts on this comment @Horta ?

Reply:
Not sure if I understand. Are you suggesting to always calculate the indices corresponding to the unique values and then decide whether to return those indices based in the provided ``return_index``? I don't have any preference if that is the case...

Reply:
IIUC the suggestion is to inline this code under the `if` on line 430.